chia/consensus/multiprocess_validation.py

```diff
@@ -220,13 +223,19 @@ async def pre_validate_blocks_multiprocessing(
             recent_blocks[curr.header_hash] = curr
             if curr.is_transaction_block:
                 num_blocks_seen += 1
+            if len(recent_plot_ids) < 4:
+                # TODO: Add `get_proof_of_space()` to BlockStore.
+                full_block = await block_store.get_full_block(curr.header_hash)
+                assert full_block is not None
+                recent_plot_ids.append(get_plot_id(full_block.reward_chain_block.proof_of_space))
```

Comment:
this is not sensitive to reorgs, right? we just look up the previous blocks every time we validate one, yeah?

Reply:
`BlockTools` stores orphans too. So if the reorg is based off an orphan, we should still find the hash.